### PR TITLE
Convert Huffman to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  }
+
+  private fun addCode(symbol: Int, code: Int, codeBitCount: Int) {
+    val terminal = Node(symbol, codeBitCount)
+
+    var accumulatorBitCount = codeBitCount
+    var node = root
+    while (accumulatorBitCount > 8) {
+      accumulatorBitCount -= 8
+      val childIndex = code.ushr(accumulatorBitCount) and 0xff
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt]:
```suggestion
      val childIndex = code ushr accumulatorBitCount and 0xff
```
To match line 176

File: okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      }
+      node = child
+    }
+
+    val shift = 8 - accumulatorBitCount
+    val start = code shl shift and 0xff
+    val end = 1 shl shift
+    for (i in start until start + end) {
+      node.children!![i] = terminal
+    }
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt]:
```kotlin
node.children!!.fill(terminal, start, start + end)
```
Avoids a null check on each loop iteration

File: okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt]: Fixed!

File: okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/http2/Huffman.kt]: Fixed!

**Pull request merged**